### PR TITLE
chore(docs): adds types column and null checks for description

### PR DESCRIPTION
**Pull request by amcdnl**
This PR:
- Adds type column for properties
- Adds null check to descriptions

### CODE REVIEW

**Comment by jelbourn** [comment]
Could you include a screenshot of how this looks? How does it fit on mobile?

**Comment by amcdnl** [comment]
Here is a before/after

img

**Comment by jelbourn** [comment]
I'm not a huge fan of how much horizontal space it takes up now; any ideas around how we can keep the name and the type in the same vertical space next to the description?

**Comment by willshowell** [comment]
From https://github.com/angular/material.angular.io/issues/287#issuecomment-335505847, would you consider moving the type to the description cell and making it distinct via font weight and padding? Just instead of "**Return Value:**", have it sat "**Type:**"

screen shot 2017-10-10 at 11 11 19 am

**Comment by jelbourn** [comment]
That could work- I do want to try emphasizing the type by using a monospace font with a different color rather than writing out `type` to see how that comes across

**Comment by amcdnl** [comment]
Here is the updated look:

img

File: tools/dgeni/templates/property.template.html [hunk 1]
```diff
@@ -25,7 +25,11 @@
     <p class="docs-api-property-name">
       {$ property.name $}
     </p>
-    <code class="docs-api-property-type">{$ property.type $}</code>
   </td>
-  <td class="docs-api-property-description">{$ property.description | marked | safe $}</td>
+  <td class="docs-api-property-description">
+    <div class="docs-api-property-type">
+      Type: <code>{$ property.type $}</code>
```
Comment (jelbourn) [inline tools/dgeni/templates/property.template.html]: I'm not a huge fan of having "Type: " explicitly. Repeated text like this creates a lot of visual noise

Comment (amcdnl) [reply tools/dgeni/templates/property.template.html]: Gotcha. Should I just remove type and leave it like is otherwise.

Comment (jelbourn) [reply tools/dgeni/templates/property.template.html]: Can you post a SS of what that looks like?

**Comment by amcdnl** [comment]
I'm not sure I like this:

img

**Comment by jelbourn** [comment]
What if we add it line with the name, TypeScript syntax style?
```
@Input()
disabled: boolean
```

**Comment by amcdnl** [comment]
@jelbourn - That doesn't look bad ->

img

**Comment by jelbourn** [comment]
I do like that more

**Comment by jelbourn** [comment]
@amcdnl can you re-create this PR? It was merged before final changes could be made

**Comment by amcdnl** [comment]
Done in #8125

**Pull request merged**